runners/java-fn-execution/src/main/java/org/apache/beam/runners/fnexecution/wire/LengthPrefixUnknownCoders.java

```diff
@@ -55,12 +56,16 @@ public static String addLengthPrefixedCoder(
     //     request to replace the coder with a byte array coder.
     //  2) the requested coder is a known coder but not a length prefix coder. In this case we
     //     rebuild the coder by recursively length prefixing any unknown component coders.
-    //  3) the requested coder is an unknown coder. In this case we either wrap the requested coder
+    //  3) the requested coder is a serialized Java coder. In this case we wrap the coder with
+    //     a length prefix coder.
+    //  4) the requested coder is an unknown coder. In this case we either wrap the requested coder
     //     with a length prefix coder or replace it with a length prefix byte array coder.
     if (ModelCoders.LENGTH_PREFIX_CODER_URN.equals(urn)) {
       return replaceWithByteArrayCoder ? lengthPrefixedByteArrayCoderId : coderId;
     } else if (ModelCoders.urns().contains(urn)) {
       return addForModelCoder(coderId, components, replaceWithByteArrayCoder);
+    } else if (CoderTranslation.JAVA_SERIALIZED_CODER_URN.equals(urn)) {
```

Comment:
This isn't what your looking for. Runners and SDKs have a duality in the coder representation. See [Encoding and Decoding Unknown Types](This isn't the correct place to fix this. It is likely that a portable runner isn't storing "two" coder representations, one for itself and one for the SDK. See [Encoding and Decoding Unknown Types](https://docs.google.com/document/d/1IGduUqmhWDi_69l9nG8kw73HZ5WI5wOps9Tshl5wpQA/edit#heading=h.sh4d5klmtfis).

The SDK may want to represent a type `Foo` but the runner can't encode or decode this type. The runner instead uses a representation which has the same wire encoding by using `LengthPrefix<Byte[]>` for its coder while the SDK would be told to use a `LengthPrefix<Foo>`. Both have the same wire representation.

Reply:
You'll want to find the location that is calling this code and see if it is using the incorrect representation (or not remembering this duality).

Reply:
Thanks for the comment. Will look into the code you mentioned!